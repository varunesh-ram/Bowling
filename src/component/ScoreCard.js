import React from 'react';
import { Constants } from '../constants/Constants';

export default class ScoreCard extends React.Component {
    header = () => {
        const headerTitles = Constants.SCORECARD_TITLES.map((title) => (
            <th key={title} colSpan="6">
                {title}
            </th>
        ));
        return headerTitles;
    };

    pinsDownOnRolls = () => {
        const rolls = [];
        for (let i = 0; i <= 20; i++) {
            const colSpanValue = i > 17 ? "2" : "3";
            rolls.push(
                <td key={i} id={"r" + i} colSpan={colSpanValue}>
                </td>
            );
        }
        return rolls;
    };

    footer = () => {
        let bottomLine = [];
        for (let i = 0; i < 11; i++) {
            bottomLine.push(
                <td key={"frame" + i} colSpan="6">
                </td>
            );
        }
        return bottomLine;
    };

    render() {
        return (
            <div className="Container">
                <table id="table" className="Scorecard" cellPadding="1" cellSpacing="0">
                    <tbody>
                        <tr>{this.header()}</tr>
                        <tr>
                            {this.pinsDownOnRolls()}
                            <td id="total-score" colSpan="6">
                            </td>
                        </tr>
                        <tr>{this.footer()}</tr>
                    </tbody>
                </table>
            </div>
        );
    }
}