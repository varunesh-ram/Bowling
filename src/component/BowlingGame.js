import React from 'react';
import ScoreCard from './ScoreCard';

export default class BowlingGame extends React.Component {
    render() {
        return (
            <div className="Game">
                <ScoreCard />
            </div>);
    }
}