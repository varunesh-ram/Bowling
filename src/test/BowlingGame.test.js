import React from 'react';
import BowlingGame from '../component/BowlingGame';
import ScoreCard from '../component/ScoreCard';
import { shallow } from 'enzyme';

describe(("<BowlingGame/> component"), () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<BowlingGame />);
    });

    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render ScoreCard component", () => {
        expect(wrapper.find(ScoreCard)).toBeDefined();
    });

});