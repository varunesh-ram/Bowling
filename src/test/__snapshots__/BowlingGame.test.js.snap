// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BowlingGame/> component should render correctly 1`] = `ShallowWrapper {}`;
